tstest/integration/integration_test.go

```diff
@@ -528,22 +530,40 @@ func (n *testNode) AwaitListening(t testing.TB) {
 	}
 }
 
-func (n *testNode) AwaitIP(t testing.TB) (ips string) {
+func (n *testNode) AwaitIPs(t testing.TB) []netaddr.IP {
 	t.Helper()
+	var addrs []netaddr.IP
 	if err := tstest.WaitFor(20*time.Second, func() error {
 		out, err := n.Tailscale("ip").Output()
 		if err != nil {
 			return err
 		}
-		ips = string(out)
+		ips := string(out)
+		ipslice := strings.Fields(ips)
+		addrs = make([]netaddr.IP, len(ipslice))
+
+		for i, ip := range ipslice {
+			netIP, err := netaddr.ParseIP(ip)
+			if err != nil {
+				return err
```

Comment:
actually I think this should t.Fatal instead of returning an error.  Returning an error inside WaitFor will just keep retrying it for 20 seconds. But if the output is bogus, we just want to fail fast.

LGTM otherwise, though.